### PR TITLE
daemontools: add manpages from Debian

### PR DESCRIPTION
**Pull request by wentasah**
###### Description of changes

Using software, whose documentation is only available online is not
always convenient. This package has only online docuemntation and
upstream provides no manpages.

This commit extends the package by manpages prepared by Debian
developers. The manpages are added to a separate output. Listing of
the outputs of this package now looks like this:

    result
    └── bin
        ├── envdir
        ├── envuidgid
        ├── fghack
        ├── multilog
        ├── pgrphack
        ├── readproctitle
        ├── setlock
        ├── setuidgid
        ├── softlimit
        ├── supervise
        ├── svc
        ├── svok
        ├── svscan
        ├── svscanboot
        ├── svstat
        ├── tai64n
        └── tai64nlocal
    result-man
    └── share
        ├── doc
        │   └── daemontools
        │       └── README.man
        └── man
            └── man8
                ├── envdir.8.gz
                ├── envuidgid.8.gz
                ├── fghack.8.gz
                ├── multilog.8.gz
                ├── pgrphack.8.gz
                ├── readproctitle.8.gz
                ├── setlock.8.gz
                ├── setuidgid.8.gz
                ├── softlimit.8.gz
                ├── supervise.8.gz
                ├── svc.8.gz
                ├── svok.8.gz
                ├── svscan.8.gz
                ├── svscanboot.8.gz
                ├── svstat.8.gz
                ├── tai64n.8.gz
                └── tai64nlocal.8.gz

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by viraptor** [comment]
Result of `nixpkgs-review pr 186354` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>daemontools</li>
  </ul>
</details>

**Review by kevincox — APPROVED** [review]

**Comment by kevincox** [comment]
Result of `nixpkgs-review pr 186354` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>daemontools</li>
  </ul>
</details>

File: pkgs/tools/admin/daemontools/default.nix [hunk 1]
```diff
@@ -32,8 +40,15 @@ stdenv.mkDerivation rec {
     for cmd in $(cat package/commands); do
       install -Dm755 "command/$cmd" "$out/bin/$cmd"
     done
+
+    tar -xz --strip-components=2 -f ${man-pages}
+    install -v -Dm644 -t $man/share/man/man8 daemontools-man/*.8
```
Comment (SuperSandro2000) [inline pkgs/tools/admin/daemontools/default.nix:45]:
```suggestion
    installManPage daemontools-man/*.8
```

Comment (wentasah) [reply pkgs/tools/admin/daemontools/default.nix]: Thanks. I updated the commit with you suggestion.

File: pkgs/tools/admin/daemontools/default.nix [hunk 2]
```diff
@@ -1,5 +1,11 @@
-{ fetchurl, bash, glibc, lib, stdenv }:
+{ fetchurl, bash, glibc, lib, stdenv, installShellFiles }:
 
+let
+  man-pages = fetchurl {
+    url = "https://salsa.debian.org/debian/daemontools/-/archive/debian/1%250.76-8/daemontools-debian-1%250.76-8.tar.gz?path=debian/daemontools-man";
```
Comment (SuperSandro2000) [inline pkgs/tools/admin/daemontools/default.nix:5]: I was thinking about this earlier but didn't have a good solution: how do we make sure those match the version of daemontools? We probably just need to watchout when updating.

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/tools/admin/daemontools/default.nix (hunk 2 above)
Comment (wentasah) [reply pkgs/tools/admin/daemontools/default.nix:5]: Given that the last release of daemontools was in 2001, I don't think we have to care too much. Watching it manually is definitely sufficient.

**Comment by kevincox** [comment]
Result of `nixpkgs-review pr 186354` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>daemontools</li>
  </ul>
</details>

**Pull request merged**